### ISSUE TITLE
Using doTrackPageView() with the PiwikTracker PHP API records server IP instead of client IP.

### ISSUE THREAD

**Issue by mahammer** [piwik/piwik#10920]
I am new to piwik so I am not sure if this is a bug, issue or feature.
- I require once the PiwikTracker.php
- Set the PiwikTracker::$URL
- create a new PiwikTracker object for my idSite
- call ->doTrackPageView()

However piwik records the server IP instead of the visitors IP. I am aware that I could use ->setIP() but then I need to auth() for what I read.
Is recording the server IP a bug or intended?

**Comment by mahammer** [comment]
I've checked the classes source and my PiwikTracker object $tracker.
`$tracker->ip == $_SERVER['REMOTE_ADDR']`
So why would `$tracker-doTrackPageView('foo')` record with `$_SERVER['SERVER_ADDR']`?

**Comment by mattab** [comment]
REMOTE_ADDR is typically set to the client address. if not, it may be you are using proxies or other configuration where IP is set in another header. you can see which IP to use by looking at the output of `var_dump($_SERVER);`

**Comment by mahammer** [comment]
You seem to have misunderstood.
On my server `$_SERVER['REMOTE_ADDR']` *is* the client IP, just like it should be.
Also the `PiwikTracker->ip` holds the client IP, just like it should do.
However, when calling `PiwikTracker->doTrackPageView()` the Server IP is recorded by Piwik insted of the client IP.

**Comment by mattab** [comment]
it should definitely work if you call setTokenAuth and pass a token with admin access to the given website in piwik. If it still does not for you, you can debug the tracking api requests see here: http://developer.piwik.org/api-reference/tracking-api#debugging-the-tracker

**Comment by mahammer** [comment]
I found the issue, I leave this here for future reference.
Your documentation is lacking or I misunderstood/overread something.
Like it said here [1] I did the following:

```
<?php
  require_once "lib/PiwikTracker/PiwikTracker.php";
  PiwikTracker::$URL="http://foo.bar.com/";
  $tracker = new PiwikTracker(1);
  $tracker->doTrackPageView("foo.dev");
?>
```
Almost everything worked like intended, `$_SERVER['REMOTE_ADDR']` and the `PiwikTracker` holding the client IP, but Piwik recorded the server IP.

The help on this suggested to find the client IP  in your `$_SERVER' and send it manually. So I did.
The documentation says that `setIp()` needs `setTokenAuth()` thus I tried:
```
<?php
  require_once "lib/PiwikTracker/PiwikTracker.php";
  PiwikTracker::$URL="http://foo.bar.com/";
  $tracker = new PiwikTracker(1);
  $tracker->setTokenAuth("XxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXx");
  $tracker->setIp($_SERVER['REMOTE_ADDR']);
  $tracker->doTrackPageView("foo.dev");
?>
```
This resulted again in Piwik recording the server IP. Which is odd, given the next step and solution.

Then I went back to the first code without setTokenAuth() and enabled debugging and saw that Piwik reports that the request lacks the rights to record CID, which equaled the client IP.

So, lastly I did this:
```
<?php
  require_once "lib/PiwikTracker/PiwikTracker.php";
  PiwikTracker::$URL="http://foo.bar.com/";
  $tracker = new PiwikTracker(1);
  $tracker->setTokenAuth("XxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXx");
  $tracker->doTrackPageView("foo.dev");
?>
```
This now properly records the client IP in Piwik. Unfortunately the documentation in [1] nowhere states that setTokenAuth() is required for this to work. Alternatively there's a bug that requires regular tracking to have admin rights.

---
[1] https://piwik.org/docs/tracking-api/#tracking-api-reference
